planning/behavior_path_planner/include/behavior_path_planner/scene_module_v2/scene_module_interface.hpp

```diff
@@ -0,0 +1,298 @@
+    }
+  }
+
+  /**
+   * @brief Called on the first time when the module goes into RUNNING.
+   */
+  virtual void onEntry() = 0;
+
+  /**
+   * @brief Called when the module exit from RUNNING.
+   */
+  virtual void onExit() = 0;
+
+  /**
+   * @brief Publish status if the module is requested to run
+   */
+  virtual void publishRTCStatus()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->publishCooperateStatus(clock_->now());
+  }
+
+  /**
+   * @brief Return true if the activation command is received
+   */
+  virtual bool isActivated()
+  {
+    if (!rtc_interface_ptr_) {
+      return true;
+    }
+    if (rtc_interface_ptr_->isRegistered(uuid_)) {
+      return rtc_interface_ptr_->isActivated(uuid_);
+    }
+    return false;
+  }
+
+  virtual void publishSteeringFactor()
+  {
+    if (!steering_factor_interface_ptr_) {
+      return;
+    }
+    steering_factor_interface_ptr_->publishSteeringFactor(clock_->now());
+  }
+
+  virtual void lockRTCCommand()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->lockCommandUpdate();
+  }
+
+  virtual void unlockRTCCommand()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->unlockCommandUpdate();
+  }
+
+  /**
+   * @brief set previous module's output as input for this module
+   */
+  void setPreviousModuleOutput(const BehaviorModuleOutput & previous_module_output)
+  {
+    previous_module_output_ = previous_module_output;
+  }
+
+  /**
+   * @brief set planner data
+   */
+  void setData(const std::shared_ptr<const PlannerData> & data) { planner_data_ = data; }
+
+  void resetPathCandidate() { path_candidate_.reset(); }
+
+  void resetPathReference() { path_reference_.reset(); }
+
+  bool isWaitingApproval() const { return is_waiting_approval_; }
+
+  bool isAlreadyApproved() const { return is_already_approved_; }
+
+  rclcpp::Logger getLogger() const { return logger_; }
+
+  PlanResult getPathCandidate() const { return path_candidate_; }
+
+  PlanResult getPathReference() const { return path_reference_; }
+
+  MarkerArray getDebugMarkers() { return debug_marker_; }
+
+  ModuleStatus getCurrentStatus() const { return current_state_; }
+
+  virtual void acceptVisitor(const std::shared_ptr<SceneModuleVisitor> & visitor) const = 0;
+
+  std::string name() const { return name_; }
+
+  std::shared_ptr<const PlannerData> planner_data_;
+
+private:
+  std::string name_;
+  rclcpp::Logger logger_;
+
+protected:
+  void updateRTCStatus(const double start_distance, const double finish_distance)
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->updateCooperateStatus(
+      uuid_, isExecutionReady(), start_distance, finish_distance, clock_->now());
+  }
+
+  virtual void removeRTCStatus()
+  {
+    if (!rtc_interface_ptr_) {
+      return;
+    }
+    rtc_interface_ptr_->clearCooperateStatus();
+  }
+
+  void alreadyApproved() { is_already_approved_ = true; }
```

Comment:
Function name should begin from verb.
So, I'd like to change the name appropriately, but I don't understand the nuance of "already".
Could you tell me the reason of this naming?

Reply:
for example, `guard`, `protect`, `block`, `interrupt` ...

Reply:
```c++
bool is_blocked_other_module_launch_;

bool islockedOtherModuleLaunch() { return is_blocked_new_module_launch_; }
void lockedOtherModuleLaunch() { return is_blocked_new_module_launch_ = true; }
void unlockedOtherModuleLaunch() { return is_blocked_new_module_launch_ = false; }
```

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/2919/commits/88d631ec4f903a21553528c3f75674ca2bfdc612.